### PR TITLE
`getCollectionContract()` helper

### PR DESCRIPTION
**Pull request by cedric25**
- Add `collectionContratAddress` to the `IExecDataProtector` class (including a default value)

- In order to simplify code for SDK's methods, create a `getContract` helper that centralizes what we need to instantiate a smart-contract wither ethers.

### CODE REVIEW

File: packages/sdk/src/dataProtector/IExecDataProtector.ts [hunk 1]
```diff
@@ -144,10 +152,7 @@ class IExecDataProtector {
     });
 
   createCollection = (): Promise<CreateCollectionResponse> =>
-    createCollection({
-      iexec: this.iexec,
-      sharingContractAddress: this.sharingContractAddress,
-    });
+    createCollection();
```
Comment (cedric25) [inline packages/sdk/src/dataProtector/IExecDataProtector.ts:155]: 🟢 SDK method signature highly simplified.

File: packages/sdk/src/dataProtector/sharing/smartContract/getCollectionContract.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { Contract } from 'ethers';
+import type { IExec } from 'iexec';
+import { ABI as collectionABI } from '../../../contracts/collectionAbi.js';
```
Comment (cedric25) [inline packages/sdk/src/dataProtector/sharing/smartContract/getCollectionContract.ts:3]: 🟢 These 3 imports are not needed anymore in SDK's business code.

File: packages/sdk/src/dataProtector/sharing/createCollection.ts [hunk 3]
```diff
@@ -1,28 +1,10 @@
-import { ethers, type Contract } from 'ethers';
-import { ABI as sharingABI } from '../../contracts/sharingAbi.js';
 import { WorkflowError } from '../../utils/errors.js';
```
Comment (cedric25) [inline packages/sdk/src/dataProtector/sharing/createCollection.ts:1]: 🟢 🟢 The file that is the most simplified.
- Less imports
- Less parameters for the function
- Less logic to instantiate the smart-contract through ethers

→ All done by the new `getCollectionContrat` helper.

**Review by Le-Caignec — APPROVED** [review]
LGTM => collection contract will removed in the future

**Pull request merged**